src/extension/src/EnableCommandHandler.py

```diff
@@ -49,8 +49,12 @@ def execute_handler_action(self):
             self.ext_output_status_handler.read_file(self.seq_no)
 
             config_settings = self.ext_config_settings_handler.read_file(self.seq_no)
-            operation = config_settings.__getattribute__(self.config_public_settings.operation)
 
+            # set activity_id in telemetry
+            if self.logger.telemetry_writer is not None:
+                self.logger.telemetry_writer.set_operation_id(config_settings.__getattribute__(self.config_public_settings.activity_id))
+
+            operation = config_settings.__getattribute__(self.config_public_settings.operation)
```

Comment:
Consider setting activity id exactly once in non-enable scenarios to a single timestamp that can be used to correlate better to all events that came out of that entire operation.